### PR TITLE
Add new extras "geo" for contacts with a locationmap

### PR DESCRIPTION
**Pull request by elioschmutz**
@jone 

You can install ftw.geo through the extra [geo]

I also removed the unused field "show_placemark".

### CODE REVIEW

File: ftw/contacts/configure.zcml [hunk 1]
```diff
@@ -41,23 +35,19 @@
         />
 
     <!-- ftw.geo -->
-    <adapter
-        for="ftw.contacts.interfaces.IContact"
-        provides="collective.geo.settings.interfaces.IGeoCustomFeatureStyle"
-        factory="collective.geo.contentlocations.geostylemanager.GeoStyleManager"
-        />
+    <configure zcml:condition="installed ftw.geo">
```
Comment (jone) [inline ftw/contacts/configure.zcml]: Please use [ftw.autofeature](https://github.com/4teamwork/ftw.autofeature) :wink:

File: ftw/contacts/configure.zcml [hunk 2]
```diff
@@ -41,23 +35,19 @@
...
-        handler=".subscribers.initializeCustomFeatureStyles"
-        />
+        <include package=".geo" />
 
-    <adapter
-        factory=".adapters.ContactLocationAdapter"
-        for="ftw.contacts.interfaces.IContact"
-        provides="ftw.geo.interfaces.IGeocodableLocation"
+        <genericsetup:registerProfile
+            name="geo"
```
Comment (jone) [inline ftw/contacts/configure.zcml]: I'd probably move the complete profile into the `geo` subpackage and name it `profile-ftw.contacts.geo:default`.
By doing that we can contain everything depending on the extra into the `geo` subpackage and load it only when the extras was used (see ftw.autofeature). What do you think?

File: ftw/contacts/profiles/geo/types/ftw.contacts.Contact.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+<object name="ftw.contacts.Contact" meta_type="Dexterity FTI"
+        i18n:domain="ftw.contacts" xmlns:i18n="http://xml.zope.org/namespaces/i18n">
+
+  <property name="behaviors" purge="False">
+      <element value="collective.geo.behaviour.interfaces.ICoordinates" />
+  </property>
```
Comment (jone) [inline ftw/contacts/profiles/geo/types/ftw.contacts.Contact.xml]: Is `IGeoreferenceable` not required? (I'm not really familiar with geo)

**Comment by jone** [comment]
Nice job :+1:

File: ftw/contacts/profiles/geo/types/ftw.contacts.Contact.xml (hunk 3 above)
Comment (elioschmutz) [inline ftw/contacts/profiles/geo/types/ftw.contacts.Contact.xml]: It seems not...

The ICoordinates is a behaviour
The IGeoreferenceable is only a markerinterface

The ICoordinates-behaviour marks the content with the IGeoreferenceable interface
```
    <plone:behavior
        title="Collective Geo Maps"
        description="This behaviour will make a content geo referenceable"
        provides=".interfaces.ICoordinates"
        marker="collective.geo.geographer.interfaces.IGeoreferenceable"
        factory=".behaviour.Coordinates"
        />
```

File: ftw/contacts/configure.zcml (hunk 2 above)
Comment (elioschmutz) [inline ftw/contacts/configure.zcml]: :+1:

**Comment by elioschmutz** [comment]
@jone :+1: done

- Add ftw.autofeature
- Move profiles to each extra-directory

**Comment by maethu** [comment]
:+1: Gogogogogo

**Comment by jone** [comment]
Since you've moved the simplelayout feature to a subpackage, you could also move parts of
- [lawgiver.zcml](https://github.com/4teamwork/ftw.contacts/blob/master/ftw/contacts/lawgiver.zcml#L17)
- [permissions.zcml](https://github.com/4teamwork/ftw.contacts/blob/master/ftw/contacts/permissions.zcml#L18)

I think it will be separated completely, then.
Looks great! :+1: :sparkles:

**Comment by elioschmutz** [comment]
@jone Thanks :smile: 

- Moved simplelayout specific lawgiver and permission configuration to the simplelayout submodule

**Comment by jone** [comment]
Very nice! It's much cleaner and now :smile: 

:construction: Waiting for https://github.com/4teamwork/bern.web/issues/523 to be fixed :wink:

**Pull request merged**